### PR TITLE
VS Code guide_installation

### PR DESCRIPTION
**Pull request by RichardHoch**
MTA 5.1.4+

Partially resolves: https://issues.redhat.com/browse/WINDUP-3103
This is the installation chapter of the guide. The introduction and the usage instructions are in https://github.com/windup/windup-documentation/pull/438 and https://github.com/windup/windup-documentation/pull/435, respectively.

Preview: https://deploy-preview-433--windup-documentation.netlify.app/docs/vs-code-extension-guide/master/index.html#installing-vs-code-extension_vsc-extension-guide

### CODE REVIEW

File: docs/topics/installing-vs-code-plugin.adoc [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+As of June 2021, VS Code has been tested on the following platforms:
+
+* Linux (Red Hat): Red Hat Enterprise Linux 7, CentOS 8, Fedora 24
+* Linux (Debian): Ubuntu Desktop 16.04, Debian 9
+* Windows 7 (with .NET Framework 4.5.2), 8.0, 8.1 and 10 (32-bit and 64-bit)
+* OS X El Capitan (10.11+)
+
+.Prerequisites
+
```
Comment (apinnick) [inline docs/topics/installing-vs-code-plugin.adoc:19]: Your prerequisites should be for the VS Code extension, not VS Code. Remember that we do not do documentation for third parties.

File: docs/topics/installing-vs-code-plugin.adoc [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+//
+// * docs/vs-code-plugin-guide/master.adoc
+
+
+[id="installing-vs-code-plugin_{context}"]
+= Installing the {VscodePluginName} extension
+
+You can install the {VscodePluginName} ({VscodePluginShortName}) extension on Linux, Windows, or macOS.
+
+As of June 2021, VS Code has been tested on the following platforms:
```
Comment (apinnick) [inline docs/topics/installing-vs-code-plugin.adoc:11]: Is this for VS Code or the extension? If it is for VS Code, we should not be documenting it.

**Comment by RichardHoch** [comment]
@apinnick: Please review. I will tag QE in Jira.

File: docs/topics/installing-vs-code-extension.adoc [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+[id="installing-vs-code-extension_{context}"]
+= Installing the {ProductShortName} extension on {VscodePluginName}
+
+You can install the {ProductShortName} extension on {VscodePluginName} on Linux, Windows or iOS.
+
+.Prerequisites
+
+* link:https://code.visualstudio.com[VS Code installed]
+* 4 GB of RAM (8 GB recommended)
+* JDK 11+
```
Comment (apinnick) [inline docs/topics/installing-vs-code-extension.adoc:15]: We do not write "later versions" with a + sign. See our style guides.

File: docs/topics/installing-vs-code-extension.adoc [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+[id="installing-vs-code-extension_{context}"]
+= Installing the {ProductShortName} extension on {VscodePluginName}
+
+You can install the {ProductShortName} extension on {VscodePluginName} on Linux, Windows or iOS.
+
+.Prerequisites
+
+* link:https://code.visualstudio.com[VS Code installed]
+* 4 GB of RAM (8 GB recommended)
```
Comment (apinnick) [inline docs/topics/installing-vs-code-extension.adoc:14]: Change to 8 GB RAM.
Never "of RAM". It's x GB RAM.

File: docs/topics/installing-vs-code-extension.adoc [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+* JDK 11+
+* `JAVA_HOME` set to `jdk11`
```
Comment (apinnick) [inline docs/topics/installing-vs-code-extension.adoc:16]: This is an environmental variable. Explain it.

File: docs/topics/installing-vs-code-extension.adoc [hunk 6]
```diff
@@ -0,0 +1,29 @@
+// Module included in the following assemblies:
+//
+// * docs/vsc-extension-guide/master.adoc
+
+
+[id="installing-vs-code-extension_{context}"]
+= Installing the {ProductShortName} extension on {VscodePluginName}
+
+You can install the {ProductShortName} extension on {VscodePluginName} on Linux, Windows or iOS.
```
Comment (apinnick) [inline docs/topics/installing-vs-code-extension.adoc:9]: Serial comma is missing.

iOS? That is Apple mobile OS and I doubt many people are installing the VS Code on their iPhone. macOS.

File: docs/topics/installing-vs-code-extension.adoc [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+.Prerequisites
+
+* link:https://code.visualstudio.com[VS Code installed]
+* 4 GB of RAM (8 GB recommended)
+* JDK 11+
+* `JAVA_HOME` set to `jdk11`
+
+.Procedure
```
Comment (apinnick) [inline docs/topics/installing-vs-code-extension.adoc:18]: WHat about MTA-CLI? When you install, the extension, the UI searches for MTA-CLI and has a button to install it. These steps have to be part of the MTA CLI. MTA CLI is NOT a usage preprequisite. It is part of the installation because it is installed through the UI before the user exits the extension. Try it and you will see what I mean.

File: docs/topics/installing-vs-code-extension.adoc [hunk 8]
```diff
@@ -0,0 +1,29 @@
+// Module included in the following assemblies:
+//
+// * docs/vsc-extension-guide/master.adoc
+
+
+[id="installing-vs-code-extension_{context}"]
+= Installing the {ProductShortName} extension on Visual Studio Code
```
Comment (apinnick) [inline docs/topics/installing-vs-code-extension.adoc:7]: I think "on" is not the right preposition. You might install an extension "in" a product. I think "for" would be a better choice.

File: docs/topics/installing-vs-code-extension.adoc [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+You can install the {ProductShortName} extension on Visual Studio Code (VS Code) on Linux, Windows, or macOS.
```
Comment (apinnick) [inline docs/topics/installing-vs-code-extension.adoc:9]: "on" > "for"

You might want to remove "on Linux, Windows, or macOS" because the user can assume that any platform that supports VS Code will support the extension.

File: docs/topics/installing-vs-code-extension.adoc [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+.Prerequisites
+
+* link:https://code.visualstudio.com[VS Code installed]
```
Comment (apinnick) [inline docs/topics/installing-vs-code-extension.adoc:13]: Unless you require a specific **version** of VS Code, I would remove it from the prerequisites. It's obvious to the user that a VS Code extension requires a VS Code installation.

File: docs/topics/installing-vs-code-extension.adoc [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+* 8 GB RAM
+* JDK 11 or later
```
Comment (apinnick) [inline docs/topics/installing-vs-code-extension.adoc:15]: Spell it out in full the first time. Also, do the same for the other installations.

File: docs/topics/installing-vs-code-extension.adoc [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+* Environmental variable `JAVA_HOME` set to `jdk11`
```
Comment (apinnick) [inline docs/topics/installing-vs-code-extension.adoc:16]: I would not call this a prerequisite because it is not something that a person is likely to have before installing the extension. This should be the first step of the installation. I'll write it for you.

File: docs/topics/installing-vs-code-extension.adoc [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+
+. In VS Code, click the *Extensions* icon on the Activity bar.
+. Enter `{ProductName}` in the Search field.
+. Select the *Migration Toolkit for Applications* extension and click *Install*.
++
+The {ProductShortName} extension icon (image:vs_MTA_extension_icon.png[MTA code extension]) is displayed on the Activity bar.
+
+[NOTE]
+====
+To uninstall the {ProductShortName} extension, click the *Extensions* icon on the Activity bar, click the Manage icon of the {ProductShortName} extension, and select *Uninstall*.
```
Comment (apinnick) [inline docs/topics/installing-vs-code-extension.adoc:28]: You have Extensions icon marked in bold and Manage icon not marked. Is there a tooltip or other indication that "Extensions" icon is called "Extensions"? If not, it should not be bold face.

Comment (apinnick) [reply docs/topics/installing-vs-code-extension.adoc:28]: I just checked. Both icons have tooltips, so "Manage" should also be in bold.

Comment (apinnick) [reply docs/topics/installing-vs-code-extension.adoc:28]: Suggest rewrite of note:
====
To uninstall the ... extension:
. In the *Extensions* view, select the *Migration Toolkit for Applications* extension.
. Click the *Manage* icon and select *Uninstall*.
====

You do not need to say "click the Extensions icon" if you specify Extensions view.
You do need to select the extension in order to access the Manage icon.
You need to spell out the name in full if you are referring to it in the UI because that is how it appears.
====

File: docs/topics/templates/document-attributes.adoc [hunk 14]
```diff
@@ -38,7 +38,7 @@
 :DocInfoProductNumber: 5.1
 :DocInfoProductNameURL: migration_toolkit_for_applications
 
-:OpenShiftProductNumber: 4.7
+:OpenShiftProductNumber: 4.8
 :ProductVersion: 5.1.4
```
Comment (apinnick) [inline docs/topics/templates/document-attributes.adoc:42]: When you rebase, this will be correct and fix the merge conflict.

File: docs/topics/installing-plugin.adoc [hunk 15]
```diff
@@ -0,0 +1,49 @@
+// Module included in the following assemblies:
```
Comment (apinnick) [inline docs/topics/installing-plugin.adoc:1]: Why is this module among your changes? It does not appear to be related to the VS Code extension guide.

Comment (RichardHoch) [reply docs/topics/installing-plugin.adoc:1]: I think it is because originally I added ifdef sections to this module with the attribute for the VS Code extension. Doing that worked fine but at one point you and I discussed it and I decided to create a separate installation module for the VS Code extension. This "added" module has been part of the PR for a while. If you think it's wiser, I can close this PR and open a new one.

Comment (apinnick) [reply docs/topics/installing-plugin.adoc:1]: That should not happen. This PR records the file as a new file, not a changed file. Check it carefully. I would not close the PR and open a new one. But I would make sure that this is not duplicating an existing file.

File: docs/topics/installing-vs-code-extension.adoc (hunk 9 above)
Comment (RichardHoch) [reply docs/topics/installing-vs-code-extension.adoc:9]: Since removing the phrase left me with "You can install the {ProductShortName} extension on Visual Studio Code (VS Code)." -- a rather uninformative sentence, I decided to go with "...on any platform that supports VS Code." WDYT?

File: docs/topics/installing-plugin.adoc (hunk 15 above)
Comment (RichardHoch) [reply docs/topics/installing-plugin.adoc:1]: The file is irrelevant, so i deleted it in the current commit. Eclipse/CSR installation is covered by eclipse-installing-plugin.adoc and VS Code installation is covered by vs-code-extension.adoc.

**Comment by RichardHoch** [comment]
@apinnick: Please review. Once the PR for the intro is changed, I'll have to rebase this branch or make other changes to its master.adoc before I squash the commits for this PR.

File: docs/topics/installing-vs-code-extension.adoc [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+.Procedure
+
+. Set the environmental variable `JAVA_HOME` to `jdk11`:
++
+[source, terminal]
+----
+$ export JAVA_HOME=jdk11
+----
+
+. In VS Code, click the *Extensions* icon on the Activity bar to open the *Extensions* view.
```
Comment (apinnick) [inline docs/topics/installing-vs-code-extension.adoc:26]: I'm not sure that Extensions view or Extensions icon are marked up. They are bold if there is a label or a tooltip. I can't recall an Extensions view label. Does it say "Extensions" somewhere when you are in the Extensions view? Do you understand what I am saying?

File: docs/topics/installing-vs-code-extension.adoc [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+. Enter `{ProductName}` in the Search field.
+. Select the *Migration Toolkit for Applications* extension and click *Install*.
++
+The {ProductShortName} extension icon (image:vs_MTA_extension_icon.png[MTA code extension]) is displayed on the Activity bar.
+
+[NOTE]
+====
+To uninstall the {ProductShortName} extension:
+
+. In the *Extensions* view, select the *{ProductName}* extension.
```
Comment (apinnick) [inline docs/topics/installing-vs-code-extension.adoc:36]: Same as above, re: Extensions view.

File: docs/topics/installing-vs-code-extension.adoc [hunk 18]
```diff
@@ -0,0 +1,38 @@
...
+. Click the *Manage* icon and select *Uninstall*.
```
Comment (apinnick) [inline docs/topics/installing-vs-code-extension.adoc:37]: Is it called the Manage icon in the UI? Also, this is worth using an image of the icon. I think it's a gear.

File: docs/topics/templates/document-attributes.adoc [hunk 19]
```diff
@@ -39,7 +39,7 @@
 :DocInfoProductNameURL: migration_toolkit_for_applications
 
 :OpenShiftProductNumber: 4.8
-:ProductVersion: 5.2.0
+:ProductVersion: 5.1.4
```
Comment (apinnick) [inline docs/topics/templates/document-attributes.adoc:42]: Version number

File: docs/topics/installing-vs-code-extension.adoc [hunk 20]
```diff
@@ -0,0 +1,38 @@
...
+= Installing the {ProductShortName} extension for Visual Studio Code
+
+You can install the {ProductShortName} extension for Visual Studio Code (VS Code) on any platform that supports VS Code.
+
+.Prerequisites
+
+* 8 GB RAM
+* JDK 11 or later
+
+
```
Comment (apinnick) [inline docs/topics/installing-vs-code-extension.adoc:16]: Remove extra line

File: docs/topics/installing-vs-code-extension.adoc [hunk 21]
```diff
@@ -0,0 +1,38 @@
+// Module included in the following assemblies:
+//
+// * docs/vsc-extension-guide/master.adoc
+
+
```
Comment (apinnick) [inline docs/topics/installing-vs-code-extension.adoc:5]: Remove extra line

File: docs/topics/installing-vs-code-extension.adoc (hunk 16 above)
Comment (RichardHoch) [reply docs/topics/installing-vs-code-extension.adoc:26]: If I'm not mistaken, you told me to bold these names 10 days ago (I've unhidden your previous comment). Did I misunderstand?

Comment (apinnick) [reply docs/topics/installing-vs-code-extension.adoc:24]: In that case, I must have checked. Otherwise I would not have told you to bold them. So you can ignore this comment. I will resolve it.

**Comment by apinnick** [comment]
> @apinnick: Please review. Once the PR for the intro is changed, I'll have to rebase this branch or make other changes to its master.adoc before I squash the commits for this PR.

@RichardHoch I don't understand.

**Comment by RichardHoch** [comment]
@apinnick -- As far as i can tell, all the issues in the PR have been resolved. Does QE need to see it again?

File: docs/vs-code-extension-guide/master.adoc [hunk 22]
```diff
@@ -11,22 +11,19 @@ include::topics/templates/document-attributes.adoc[]
 = Visual Studio Code Extension Guide
```
Comment (apinnick) [inline docs/vs-code-extension-guide/master.adoc]: I thought you were going to change this to sentence case.

File: docs/topics/installing-vs-code-plugin.adoc [hunk 23]
```diff
@@ -0,0 +1,35 @@
...
+You can install the {ProductShortName} extension for Visual Studio Code (VS Code) on any platform that supports VS Code.
+
+.Prerequisites
+
+* 8 GB RAM
+* JDK 11 or later
+
+.Procedure
+
+. Set the environmental variable `JAVA_HOME` to `jdk11`:
```
Comment (apinnick) [inline docs/topics/installing-vs-code-plugin.adoc]: You can remove "to jdk11" at the end of the lead-in sentence.

File: docs/topics/installing-vs-code-plugin.adoc [hunk 24]
```diff
@@ -0,0 +1,35 @@
+// Module included in the following assemblies:
+//
+// * docs/vsc-extension-guide/master.adoc
+
+[id="installing-vs-code-extension_{context}"]
+= Installing the {ProductShortName} extension for Visual Studio Code
+
+You can install the {ProductShortName} extension for Visual Studio Code (VS Code) on any platform that supports VS Code.
```
Comment (apinnick) [inline docs/topics/installing-vs-code-plugin.adoc]: I'm not sure we need "on any platform".... I think it's probably enough to say "You can install the x extension for y."

File: docs/topics/installing-vs-code-plugin.adoc [hunk 25]
```diff
@@ -0,0 +1,35 @@
...
+----
+
+. In VS Code, click the *Extensions* icon on the Activity bar to open the *Extensions* view.
+. Enter `{ProductName}` in the Search field.
+. Select the *Migration Toolkit for Applications* extension and click *Install*.
++
+The {ProductShortName} extension icon (image:vs_MTA_extension_icon.png[MTA code extension]) is displayed on the Activity bar.
+
+[NOTE]
+====
```
Comment (apinnick) [inline docs/topics/installing-vs-code-plugin.adoc]: Honestly, I am not sure we need this note. The MTA extension is uninstalled just like any other VSC extension. That procedure is documented in the VSC documentation.

File: docs/vs-code-extension-guide/master.adoc (hunk 22 above)
Comment (RichardHoch) [reply docs/vs-code-extension-guide/master.adoc]: Was planning to leave alone for and later do all documents. No need to wait, though.

Comment (apinnick) [reply docs/vs-code-extension-guide/master.adoc]: I thought you said that sentence case was already being used in the attributes for the titles.

**Comment by apinnick** [comment]
LGTM
